### PR TITLE
Realized cash from invoices

### PR DESCRIPTION
**Pull request by mbayopanda**
This report add a way to know from invoices made what cash was paid and what is remaining.

### CODE REVIEW

File: client/src/i18n/en/report.json
Comment (jniles) [inline client/src/i18n/en/report.json]: "Remains" on its own is usually what you call a corpse ☠️ .  Go ahead and make this "View details about remaining debts".

File: client/src/i18n/en/tree.json
Comment (jniles) [inline client/src/i18n/en/tree.json]: _Technically speaking_, I think this should be "Realized Income" or "Realized Revenue".   The term "profit" is what you make after expenses are taken out.

File: server/controllers/finance/reports/realized_profit/index.js
Comment (jniles) [inline server/controllers/finance/reports/realized_profit/index.js:105]: Just FYI - I don't think you need to do a `DROP TABLE` query when you are using a transaction - transactions _should_ run in isolated memory environments and they should clean up any temporary tables at the end of execution.

File: server/controllers/finance/reports/realized_profit/index.js
Comment (jniles) [inline server/controllers/finance/reports/realized_profit/index.js:102]: Hmm.. it looks like you always put the `groupByServiceAndDebtorGroup` in the query.  Since that is the case, go ahead and write it as part of the `tableQuery`.  .e.g :
```sql
      SELECT
        SUM(w.paid) paid, SUM(w.invoiced) invoiced, SUM(w.invoiced - w.paid) remaining,
        w.debtorGroupName, w.serviceName, w.service_id, w.debtor_group_uuid 
      FROM (${globalQuery})w
      GROUP BY w.service_id, w.debtor_group_uuid;
```

**Review by jniles — APPROVED** [review]
@mbayopanda, this is really clean.  Your code is super easy to read and elegant.  I haven't tested it for performance issues, but we can always come back and try to improve it later.  Well done!

I've made a few minor comments to translation and variable declarations.  There isn't a need for a second review - once you make those changes, we can go ahead and merge the code.

File: server/controllers/finance/reports/realized_profit/index.js
Comment (mbayopanda) [reply server/controllers/finance/reports/realized_profit/index.js:102]: The `tableQuery` without `groupByServiceAndDebtorGroup` is used for getting totals.

File: server/controllers/finance/reports/realized_profit/index.js
Comment (mbayopanda) [reply server/controllers/finance/reports/realized_profit/index.js:105]: Without the `DROP TABLE` there is an error which is thrown

File: server/controllers/finance/reports/realized_profit/index.js
Comment (jniles) [reply server/controllers/finance/reports/realized_profit/index.js:102]: Aha, didn't see that.  No problem then.

File: server/controllers/finance/reports/realized_profit/index.js
Comment (jniles) [reply server/controllers/finance/reports/realized_profit/index.js:105]: :'(.  Everything I used to believe is a lie.  Thank you for correcting me!

**Review by jniles — APPROVED** [review]
Looks good to me.

bors r+

**Pull request merged**
